### DISCUSSION TITLE
Valetudo 2023.02.0

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1709] [Release Discussions]
<div align="center">
    <img src="https://raw.githubusercontent.com/Hypfer/Valetudo/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2023.02.0</h2></p>
</div>

Custom instance names, more robots and quality of life improvements for existing robots

## Custom instance name

It is now possible to set a custom name for your Valetudo instance.

![image](https://user-images.githubusercontent.com/974410/218273944-027b6a14-50fd-4174-b11a-448316135a5b.png)

This should be especially helpful for setups with multiple floors and thus multiple robots.

[<img src="https://user-images.githubusercontent.com/974410/218273978-aff0d895-6089-490b-877d-0d6b5b8f3e2d.png" width=300>](https://user-images.githubusercontent.com/974410/218273978-aff0d895-6089-490b-877d-0d6b5b8f3e2d.png)

Additionally, this name is also part of the data advertised via MQTT, meaning that the robot will show up like that in Home Assistant as well.

### Mi Robot Vacuum-Mop 2 Ultra

After buying a `Mi Robot Vacuum-Mop 2 Ultra` to evaluate if it can be rooted and supported, I can now answer yes to both questions.

![image](https://user-images.githubusercontent.com/974410/218274059-6df8c29e-5da3-4537-b900-3e338efb8ecf.png)

It's like a Dreame Z10 Pro but with the auto-empty dock being optional and a ToF sensor instead of line lasers for obstacle avoidance.

### More Viomis

Valetudo users Matthias and @Zombielinux meanwhile tried the CRL200S rooting instructions with the `Viomi SE` and `Wyze Robot Vacuum` and succeeded in doing so. 

Unfortunately though, shortly after that, another user reported that they've bricked their `Wyze Robot Vacuum` while attempting to root it :(
Thus, for now only the `Viomi SE` will be part of the Supported Robots list until we gain further insight into the issue.

![image](https://user-images.githubusercontent.com/974410/218274073-d869ebaa-fe01-40d0-95a3-38943c866795.png)

### Roborock S7

In the last month, I've also bought a `Roborock S7`. Using this test device, I was able to work through the rooting procedure, fixing things along the way. Apart from that, Quirks for a few additional S7 features have been added to Valetudo as well.

![image](https://user-images.githubusercontent.com/974410/218274228-f08457f9-dc98-4693-8957-3846336ffca8.png)

As of now, there are no more known issues with the S7 excluding the rather complicated disassembly that is **unsuited for beginners**.
It is now considered supported by Valetudo and rootable with the latest firmware.

### Roborock General

People with a Roborock supporting the `MappingPassCapability` should now instantly see segments after the mapping pass has finished.

For owners of a Roborock that needs to do a full cleanup to split the map into segments, a new Quirk was added that allows to manually trigger the split. While janky in implementation, it can nonetheless be helpful for quick remappings using for example the go-to feature.

![image](https://user-images.githubusercontent.com/974410/218274248-d67a2291-c84e-4dbb-a9a5-4e1c35a85b2a.png)

![image](https://user-images.githubusercontent.com/974410/218274251-dd3a38fb-2933-48e2-b38e-ee56f599bf6f.png)

### Segments on vSLAM Dreames

Segments should now appear instantly after a full cleanup on vSLAM Dreames such as the 1C, 1T or F9. Previously it was necessary to trigger a second cleanup for that.

![image](https://user-images.githubusercontent.com/974410/218274285-999f45d7-4345-4bcc-ac93-2225e2487449.png)

## Misc 

1. The Live Map now remembers the previously selected mode in the browser's local storage.

2. Valetudo now pings the Gateway every 15 Minutes to keep the Wi-Fi module alive.
This should help in situations where the Webinterface became unresponsive after being idle for a few hours.

3. There's now no more mop attachment reminder on Dreame robots that feature automatic mop drying.

## The usual

If you want to see Valetudo on more robots and/or like this release, you might want to consider donating:

[https://github.com/sponsors/Hypfer](https://github.com/sponsors/Hypfer)

[https://builder.dontvacuum.me/donations.txt](https://builder.dontvacuum.me/donations.txt)

### Breaking Changes

- Allow setting a custom friendly name for the valetudo instance [`612c484`](http://github.com/Hypfer/Valetudo/commit/612c484d403969d33cdc1eeb560ee61cc8624bb0)

### Features

- Migrate existing MQTT friendlyName to instance friendlyName on upgrade [`3d9c7c2`](http://github.com/Hypfer/Valetudo/commit/3d9c7c2dfb02a3562233dbcd83225435c40b77b2)
- **vendor.roborock**: Add manual map segment trigger quirk [`54891dc`](http://github.com/Hypfer/Valetudo/commit/54891dc7f64a1217a799c7fbe387f11fb93fe653)
- **ui**: Replace consumables view with boring but unified ListMenu [`7173bc6`](http://github.com/Hypfer/Valetudo/commit/7173bc6158c331b1b0e208161a61d86807411d41)
- **vendor.dreame**: P2150 (Mi Robot Vacuum-Mop 2 Ultra) support [`1909f24`](http://github.com/Hypfer/Valetudo/commit/1909f24a1c416ab1ae33f407c4bc2fe3ad66a929)
- **vendor.roborock**: More quirks for the S7 [`f789df0`](http://github.com/Hypfer/Valetudo/commit/f789df03fec6d920655901cd5fa819217b25b9e6)
- Periodically ping the gateway when embedded to keep the wifi module alive [`d2637fd`](http://github.com/Hypfer/Valetudo/commit/d2637fd7a74b4c6009f63101946318e18496b7df)
- **ui**: Remember the last selected live map mode in the browser [`cf15896`](http://github.com/Hypfer/Valetudo/commit/cf158963b9cfc5245faa3f159848b199e85d91db)
- **vendor.viomi**: Add mop pattern quirk to Viomi V7 (#1682) [`127a086`](http://github.com/Hypfer/Valetudo/commit/127a0865e94f219afc1ab5c977061bc3198ae20a)

### Fixes

- **vendor.dreame**: Don't raise the mop attachment reminder for robots that can dry their own mops [`aba7931`](http://github.com/Hypfer/Valetudo/commit/aba7931b4a464b313bab0a24ca8e2bba8a366637)
- **vendor.dreame**: Display vSlam segments after initial cleanup [`a00280a`](http://github.com/Hypfer/Valetudo/commit/a00280a87479e71d31f15751ee558865b011de56)
- **vendor.roborock**: Fix map not immediately showing segments after a mapping pass [`25a6670`](http://github.com/Hypfer/Valetudo/commit/25a6670caf5dd8253c9f8fa4a278d49dfd2f62a5)
- **vendor.dreame**: Newer 1T firmwares have a sensor consumable [`ad59e09`](http://github.com/Hypfer/Valetudo/commit/ad59e09a1aca1f23c79f1ba89631326443247d2c)
- **webserver**: Add return after failing to load OpenAPI spec (#1689) [`b37fb09`](http://github.com/Hypfer/Valetudo/commit/b37fb0918c1b57cee52724fc3edf793034b4f605)
- **vendor.viomi**: Fix manual control state not being reported correctly [`3fe2998`](http://github.com/Hypfer/Valetudo/commit/3fe2998dfecc1cee7545ec57c543e23cda7f2308)
- Handle special character ssids in linux wifi status [`d75514b`](http://github.com/Hypfer/Valetudo/commit/d75514bd5b49a1777a8d4b09554bd6fcde2702e7)

### Chores

- **release**: 2023.02.0 [`6b01ed5`](http://github.com/Hypfer/Valetudo/commit/6b01ed5d182b5c0cffa9a4349b481f8fc2bc7879)
- Update actions to fix github workflow deprecation warnings [`d4a7ae3`](http://github.com/Hypfer/Valetudo/commit/d4a7ae3e38daf7ef7c26f51e4e9f62f9df702600)
- Formatting [`bebf0a4`](http://github.com/Hypfer/Valetudo/commit/bebf0a4cc0921f3dd3c5cdfb0875ffa77743e3c3)
- Remove config migration code [`50a15a0`](http://github.com/Hypfer/Valetudo/commit/50a15a06c20dd3b4c28a0c61b26a8d09f8f7195d)
- Delete obsolete deployment instructions [`beee8b1`](http://github.com/Hypfer/Valetudo/commit/beee8b17d3b44fa9f939fab5cee7008de8e06d24)

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2023.02.0'>Valetudo 2023.02.0</a>.</em>

**Comment by zombielinux** [comment]
I’d love to know more about the Wyze being bricked. 

I know there were some that were unusable out of the box from Wyze. Those were supposedly “bricked” from the start.

**Reply by Hypfer** [reply]
Similar to the Viomi v8, it seems that there are robots with the same model name that behave differently.

In this instance, the "press power button 10x" method did not work, however the enable-adb.sh script did, as there was an adbd that very briefly appeared.
What this script does is that it disables the main process as that would - at least on viomi v7 - disable adb access as soon as it starts up: https://github.com/Hypfer/valetudo-crl200s-root/blob/ab56659fa8d36d039fb6bb9611760793ec0f0e42/enable-adb.sh#L70-L72

This effectively bricks the robot but on the v7 that's fine because then you'll have adb access to undo that.
With the robot of that user however, adb never came back :/
This likely means that the whole business of starting and stopping adb works differently in that wyze firmware

**Comment by Skjivar** [comment]
Hey
I tried to flash my vacuum cleaner Roborock S7 with the image from Dustbuilder. However, after flashing I could not access the robot via ssh. 
You wrote that you were able to flash your S7, I would be interested to know what software version your robot had at that time. Or do you know if flashing should work with version 2.15.78?

Even though this is not a direct dustbuilder discussion I hope it still fits the topic here.
Thanks a lot

**Reply by Hypfer** [reply]
Indeed off-topic for these release notes. It would be best to discuss this in one of our telegram groups.

Rooting should work for all known firmware versions including 1578 and 1612

**Comment by elitistphoenix** [comment]
"a new Quirk was added that allows to manually trigger the split"
This was awesome and incredibly useful. Thank you!

**Comment by rufik** [comment]
Hmmm, there is no option in UI to choose how many times my Viomi SE should cleanup segment or area. Is it by seeing or bug?

![Screenshot_20230225-114308](https://user-images.githubusercontent.com/425558/221352937-60f65ed5-4d60-4736-8dcc-d8865062621f.png)

**Reply by Hypfer** [reply]
To my knowledge, that feature is not supported by the viomi firmware

**Reply by rufik** [reply]
Well, MiHome app has an option to repeat vacuuming and it works. Do you thing this feature is app-controlled, no fw-controlled?

**Reply by Hypfer** [reply]
As I haven't ever used the app, I have to rely on footage on youtube here.
I guess you mean the "secondary cleanup" toggle as can be seen in this screenshot?

![image](https://user-images.githubusercontent.com/974410/221363663-e0ada759-38e8-48a6-ba7b-b61254390ddd.png)

Because there at least don't seem to be any controls for that in the map view itself.

That works different from how it is with the other robots but it should be possible to implement.
My only concern there is that it requires a second command to be executed first, which given the unreliable nature of the viomi firmware might set that flag even though it shouldn't be.

We'll see if someone complains
https://github.com/Hypfer/Valetudo/commit/72e6b84f070be8b7e90b7987d8ac250d4d0b4618

**Reply by rufik** [reply]
Yes, it is "secondary cleanup" indeed.
Thanks a lot, I'd be happy to test is once it's released :)

**Reply by rufik** [reply]
I've tried this feature (in 2023.03.0) and it looks like working very well 👍 Thank you very much!

**Comment by Biont** [comment]
Our _Dreame L10 Pro_ has been running Valetudo for many months and it has been a joy!
After applying this update though, it is very unresponsive: the frontend rarely loads, MQTT does not respond and I cannot even ssh into it anymore.

Sometimes a reboot helps, but I did not have any luck this evening. It seems to power down and reboot normally and the wifi LED comes up, but that's all I'm getting.

Please do you have any advice how to troubleshoot this? I will try to connect with UART some time in the next days. Would you know what to check if I can access the system that way?

Thank you.

**Comment by TheDiddel** [comment]
Hello everyone, since the last update (2023.02.0) the mode can no longer be set - "vacuum" is the only mode that works. I get the following error message: PresetSelectionCapabilityRouter: Received error from robot while handling route "/preset" { body: { name: 'vacuum_and_mop' }, message: 'Error code -1' }
System information:
Dreame L10 Pro, Firmware Version 1105 Release: 2023.02.0
I have already tried the nightly version, but unfortunately this does not change the problem,
First of all, thank you very much for your feedback and basically: Thank you very much for your work! 
Best regards
Mike
